backend/canvas_app_explorer/views.py

```diff
@@ -23,6 +23,10 @@ class LTIToolViewSet(viewsets.ViewSet):
     lookup_url_kwarg = 'canvas_id'
 
     def list(self, request: Request) -> Response:
+        if not 'course_id' in request.session:
+            msg_miss_course_id = "Cannot find course id in session."
+            logger.warn(msg_miss_course_id)
+            return Response(msg_miss_course_id)
```

Comment:
We could split the public view into another API endpoint and view. Then the frontend could change the logic based on authentication info we put into the context processor. I'm not exactly sure where the branching should happen, but the frontend would need to know that it's not going to get `navigation_enabled` data.

Reply:
@jonespm @ssciolla so should we wait for #147 implementation, and maybe close #178 as it is no longer relevant?

Reply:
Yeah, I think this issue will go away once #147 is resolved. I think there's also a separate/side discussion happening here also about how best to implement #27. I'm envisioning a separate API view/URL for the public data, rather than branching within this view. We may have to do it that way so we can withhold permission restrictions.

Reply:
Nice! I will resolve this conversation and close #178.